core/src/main/java/org/apache/iceberg/BaseIncrementalScan.java

```diff
@@ -35,6 +35,26 @@ protected BaseIncrementalScan(
   protected abstract CloseableIterable<T> doPlanFiles(
       Long fromSnapshotIdExclusive, long toSnapshotIdInclusive);
 
+  @Override
+  public ThisT fromSnapshotInclusive(long fromSnapshotId, String branch) {
+    Preconditions.checkArgument(
+        table().snapshot(fromSnapshotId) != null,
+        "Cannot find the starting snapshot: %s",
+        fromSnapshotId);
+
+    Snapshot branchSnapshot = table().snapshot(branch);
+    Preconditions.checkArgument(branchSnapshot != null, "Cannot find the branch: %s", branch);
```

Comment:
in `BaseTableScan` we say `Cannot find ref %` so it would be good to align this part here as well. Also it would be good to use `ref` or `referenceName` rather than `branch` across parameters/variable names, since that is more precise imo